db/migrate/20231026010725_create_posts.rb

```diff
@@ -2,13 +2,13 @@ class CreatePosts < ActiveRecord::Migration[7.1]
   def change
     create_table :posts do |t|
       t.string :title
-      t.text :text
-      t.integer :comments_counter
-      t.integer :likes_counter
-      t.integer :author_id, null: false
-      
+      t.text :body
+      t.references :author, null: false, foreign_key: { to_table: :users }
+      t.integer :comments_counter, default: 0
+      t.integer :likes_counter, default: 0
       t.timestamps
     end
-    add_foreign_key :posts, :users, column: :author_id
+
+    add_index :posts, :author_id
```

Comment:
Hello @Nessrine88,
I'm currently reviewing your project and your migration is not running because you have to remove the Redundant Index Creation**: Since Rails takes care of creating an index for `t.references`, you can safely remove the `add_index :posts, :author_id` line from your `CreatePosts` migration.
Your migration should look like this after the change:
```ruby
class CreatePosts < ActiveRecord::Migration[7.1]
  def change
    create_table :posts do |t|
      t.string :title
      t.text :body
      t.references :author, null: false, foreign_key: { to_table: :users }
      t.integer :comments_counter, default: 0
      t.integer :likes_counter, default: 0
      t.timestamps
    end
    # Removed add_index :posts, :author_id
  end
end
```
Please make this adjustment, and your migrations should run smoothly.
Keep up the good work!